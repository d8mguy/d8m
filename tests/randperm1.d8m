import go "math/rand"
given(T::entity) {
    val randperm = \mod(lst : list(T), randSrc: \(integer)->integer) {
        var lim = count(lst)
        each(i^0...count(lst)-1) {
            val j = i+randSrc(lim)
            swap(lst, i, j)
            lim -= 1
        }
        lst
    }
}
val rptest = \imp() {
    val randx = rand.New(rand.NewSource(1))
    val rsrc = \(lim:integer) { randx.Intn(lim) }
    var ints = [0,1,2,3,4,5,6,7,8,9]
    each(i^1..20) println(randperm(ints, rsrc))
}
rptest()
