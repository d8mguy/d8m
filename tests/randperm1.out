[1, 7, 9, 3, 5, 8, 0, 2, 4, 6]
[1, 0, 6, 2, 9, 4, 8, 3, 5, 7]
[3, 8, 7, 4, 0, 5, 6, 2, 9, 1]
[9, 2, 1, 4, 0, 8, 7, 5, 3, 6]
[5, 3, 9, 1, 8, 0, 2, 4, 6, 7]
[4, 6, 2, 7, 5, 8, 0, 9, 1, 3]
[4, 1, 2, 0, 8, 5, 3, 9, 7, 6]
[3, 2, 4, 0, 1, 7, 5, 8, 9, 6]
[4, 5, 6, 8, 7, 9, 2, 3, 0, 1]
[8, 0, 3, 5, 9, 4, 1, 2, 7, 6]
[7, 4, 9, 8, 3, 5, 1, 2, 6, 0]
[6, 3, 0, 7, 9, 2, 8, 1, 4, 5]
[6, 0, 9, 7, 2, 1, 8, 4, 5, 3]
[4, 2, 5, 1, 9, 8, 3, 0, 6, 7]
[1, 8, 2, 0, 9, 5, 4, 7, 3, 6]
[6, 1, 9, 4, 7, 8, 5, 3, 2, 0]
[5, 6, 1, 4, 0, 9, 8, 2, 3, 7]
[0, 4, 7, 8, 5, 2, 3, 6, 1, 9]
[1, 6, 0, 9, 5, 7, 8, 2, 4, 3]
[1, 3, 0, 4, 9, 2, 6, 7, 8, 5]
